### PR TITLE
Lti autobind v2

### PR DESCRIPTION
**Pull request by AugustH**
- only active when plugin is loaded
- binds the lti-provided user to the course automatically

### CODE REVIEW

**Comment by GuillaumeDerval** [comment]
It's not really a plugin as it tightly integrates in the core of INGInious. It seems possible to only keep some new hooks and do everything in the plugin. We think it would be better. We would gladly merge these new hooks :-)

**Comment by AugustH** [comment]
new try: new hook + everything moved into plugin

**Review by anthonygego — CHANGES_REQUESTED** [review]
@GuillaumeDerval We finally never had your final thoughts on this.

File: inginious/frontend/plugins/ltiautobind/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    lti_email = data['email']
+    lti_name = data['realname']
+    lti_id = lti_email[0:lti_email.find('@')]
+    user_profile = ltipage.database.users.find_one({"username": lti_id})
+    if not user_profile:
+        # New user, create an account using email address
+        user_profile = {"username": lti_id,
+                        "realname": lti_name,
+                        "email": lti_email,
+                        "bindings": {data["consumer_key"]: [lti_id, {}]},
```
[the remark this answers is not in the document]
Comment (AugustH) [reply inginious/frontend/plugins/ltiautobind/__init__.py:12]: Seems the core concept is ok - i will prepare a new patch shortly

[the remark this answers is not in the document]
Comment (AugustH) [reply inginious/frontend/plugins/ltiautobind/__init__.py:12]: I don't understand the "ltibindings" -- with it i get errors in frontend/user_manager.py line 333 which uses ``info["bindings"]``.

And if i want the user to be able to login the binding should include the id of the auth method (eg. the ldap plugin used).

[the remark this answers is not in the document]
Comment (anthonygego) [reply inginious/frontend/plugins/ltiautobind/__init__.py:12]: Indeed the bindings entry should exist and be an empty dictionary if the account does not exist. It is used for external authentication methods. So you need both ``bindings`` and ``ltibindings``.

**Comment by AugustH** [comment]
created new version, see pull request #620

**Comment by Drumor** [comment]
Closed as there is a new version of the PR

**Pull request closed without merging**